moto/batch/responses.py

```diff
@@ -0,0 +1,296 @@
+from __future__ import unicode_literals
+from moto.core.responses import BaseResponse
+from .models import batch_backends
+from six.moves.urllib.parse import urlsplit
+
+from .exceptions import AWSError
+
+import json
+
+
+class BatchResponse(BaseResponse):
+    def _error(self, code, message):
+        return json.dumps({'__type': code, 'message': message}), dict(status=400)
+
+    @property
+    def batch_backend(self):
+        """
+        :return: Batch Backend
+        :rtype: moto.batch.models.BatchBackend
+        """
+        return batch_backends[self.region]
+
+    @property
+    def json(self):
+        if self.body is None or self.body == '':
+            self._json = {}
+        elif not hasattr(self, '_json'):
+            try:
+                self._json = json.loads(self.body)
+            except json.JSONDecodeError:
+                print()
+        return self._json
+
+    def _get_param(self, param_name, if_none=None):
+        val = self.json.get(param_name)
+        if val is not None:
+            return val
+        return if_none
+
+    def _get_action(self):
```

Comment:
I'm working on the mock for AWS IoT, and batch and iot both have same api-protocol, 'rest-json'. But the uri of batch is not *rest-full* at all...